### ISSUE TITLE
Timestamp is perhaps not converted into milliseconds as claimed by a comment 

### ISSUE THREAD

**Issue by euler314** [matteocontrini/sms-wp-to-android#42]
On [line 96](https://github.com/matteocontrini/sms-wp-to-android/blob/master/convert.py#L96) a conversion from "Windows file timestamp" to "UNIX milliseconds timestamp". The claim remains unclear, and at least causes problems when used together with the SMS Backup and Restore software. In particular, the SMS Backup and Restore software handles incorrectly old enough (i.e., short enough timestamp strings) causing dates to appear incorrectly on the messages. "Short enough timestamps" correspond to at least dates from the late 2007.

This can be fixed by replacing  [line 96](https://github.com/matteocontrini/sms-wp-to-android/blob/master/convert.py#L96) by:

    ts = (int(m.find('LocalTimestamp').text) / (10 * 1000 * 1000) - 11644473600) * 1000

The issue might not be strictly related to the script but to the SMS Backup and Restore software, and some extra care is needed if used together with the recent "MMS conversion feature". Here, to avoid out of range errors for the timestamp, one should retain the original line 96, and instead for [line 203](https://github.com/matteocontrini/sms-wp-to-android/blob/master/convert.py#L203) do

    date=ts*1000,

and also for [line 102](https://github.com/matteocontrini/sms-wp-to-android/blob/master/convert.py#L102) do:

    timestamp=ts*1000,

For me, MMS messages were converted incorrectly without these modifications (in particular, message dates were reported incorrectly). With the line edits as above in place, the conversion worked flawlessly with the SMS Backup and Restore.

Again, this could be helpful for somebody else and should be reported somewhere even if the script itself is not modified (but I don't perhaps see a reason to not modify it accordingly).

**Comment by matteocontrini** [comment]
Thanks for the report. A PR would be appreciated for this as well :)

**Comment by sschuberth** [comment]
FYI, https://github.com/matteocontrini/sms-wp-to-android/pull/43/commits/97bd92becfb6fa912f0ba263746d71ce3b67e675 basically undoes my https://github.com/matteocontrini/sms-wp-to-android/pull/35/commits/d6866f74075602660c182fc8b45cf49478b25b2f which I did because for me dates were wrong when converting MMS (and I don't believe SMS use different timestamps). Also, my code resembles the snippets I found throughout the net, e.g. in the article at http://www.bluethinking.com/?p=167. It's odd that it caused problems for you, @euler314.

**Comment by matteocontrini** [comment]
Could it be that you are using different versions of the SMS backup app (the Windows one)?
When I transferred my messages almost 2 years ago I didn't have problems with dates conversions, and I was using the app on a Windows Phone 8.1 phone.
I'm not sure but it might be that the Windows 10 Mobile version of the app is different. I don't remember if it was redesigned in the meanwhile...

**Comment by sschuberth** [comment]
> Could it be that you are using different versions of the SMS backup app (the Windows one)?

Even if so, I don't think it would make a difference, because the timestamps are Windows file system timestamps, and I don't believe any version of the Windows app would arbitrarily multiply that number by 1000 instead of using it as-is.

**Comment by matteocontrini** [comment]
You're right, but it appears there are multiple algorithms for converting Windows timestamps to UNIX timestamps. Your d6866f74075602660c182fc8b45cf49478b25b2f changed something but many people (including me) used the script before that change without having problems with dates.

It's getting quite mysterious.

**Comment by sschuberth** [comment]
> It's getting quite mysterious.

Indeed. Which is why I was a bit surprised that you did not have more questions when reviewing my change back then ;-)

**Comment by matteocontrini** [comment]
I only remember that I spent some time trying to get it working properly :P 

Anyway, the only difference between your version and the previous version is actually that one result is in milliseconds and the other one in seconds.

Trying with this Windows timestamp:
```js
original = 130853567488322147
```

The original conversion algorithm (since day one) gives:
```js
original / 10000 - 11644473600000
```
```js
1440883148832.2148
```

The "new" algorithm, introduced with #35, reversed with #43 gives:

```js
original / (10 * 1000 * 1000) - 11644473600
```
```js
1440883148.8322144
```

Looking at a recent backup made with the [SMS Backup & Restore app](https://play.google.com/store/apps/details?id=com.riteshsahu.SMSBackupRestore), it seems that timestamps are in milliseconds, so the original version should be the right one. @sschuberth which problems were you having specifically before making the change in #35? Was it related to MMS messages only perhaps?

**Comment by sschuberth** [comment]
> Was it related to MMS messages only perhaps?

I cannot tell for sure as I only tried with MMS, but not with SMS. IIRC the problem I had was that I got some sort of integer overflow when converting some MMS. So I was double-checking the conversion code and found it to be a factor of 1000 too large compared to the code I found on the net. I tried with the code I found on the net, the overflow went away, and the date of the MMS seem to be correct, so I was satisfied.
